### PR TITLE
Adding support for multi-NIC network in GKE cluster and node pool.

### PR DESCRIPTION
**Pull request by soumyapani**
If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes https://github.com/hashicorp/terraform-provider-google/issues/14295"  in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note: enhancement
container: added multi-NIC network for GKE cluster and node pool.
```

### CODE REVIEW

**Comment by c2thorn** [comment]
Got this error on 
```
  | Error: googleapi: Error 400: Enable_multi_networking is supported only with advanced datapath.
  | Details:
  | [
  |   {
  |     "@type": "type.googleapis.com/google.rpc.RequestInfo",
  |     "requestId": "[redacted]"
  |   }
  | ]
  | , badRequest
  | 
  |   with google_container_cluster.cluster,
  |   on terraform_plugin_test.tf line 61, in resource "google_container_cluster" "cluster":
  |   61: resource "google_container_cluster" "cluster" {
 ```

**Comment by soumyapani** [comment]
Looks like we are able to create the node pool but it failed with below error
```
vcr_utils.go:152: Step 1/2 error: Error running apply: exit status 1
        
        Error: NodePool tf-test-np-yxyquqiofi-mutli-nic was created in the error state "ERROR"
        
          with google_container_node_pool.with_multi_nic,
          on terraform_plugin_test.tf line 79, in resource "google_container_node_pool" "with_multi_nic":
          79: resource "google_container_node_pool" "with_multi_nic" {
```

Can we please check if it is due to a2 machine availability ?

File: mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb [hunk 1]
```diff
@@ -317,6 +317,30 @@ func TestAccContainerCluster_withILBSubsetting(t *testing.T) {
...
+
+	clusterName := fmt.Sprintf("tf-test-cluster-%s", acctest.RandString(t, 10))
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy: testAccCheckContainerClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerCluster_enableMultiNetworking(clusterName),
```
Comment (c2thorn) [inline mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb:332]: can we add another set of steps where we update the new fields, to test their updatability?

File: mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb [hunk 2]
```diff
@@ -521,6 +521,33 @@ func TestAccContainerNodePool_withNetworkConfig(t *testing.T) {
...
+	np := fmt.Sprintf("tf-test-np-%s", randstr)
+	network := fmt.Sprintf("tf-test-net-%s", randstr)
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckContainerClusterDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccContainerNodePool_withMultiNicNetworkConfig(cluster, np, network),
```
Comment (c2thorn) [inline mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb:539]: same here, let's test their updatability with another config and import step.

**Review by c2thorn — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb (hunk 2 above)
Comment (soumyapani) [reply mmv1/third_party/terraform/tests/resource_container_node_pool_test.go.erb:539]: Node pools with multi nic cannot be updated. They need to be destroyed and recreated without multi nic. As discussed offline I will test it just for the boolean flag.

**Comment by c2thorn** [comment]
I'm thinking the failure here was intermittent, let me try it again.

**Review by c2thorn — APPROVED** [review]

**Pull request merged**